iib/workers/tasks/build.py

```diff
@@ -105,42 +105,54 @@ def _cleanup():
 
 
 @retry(exceptions=IIBError, tries=3, logger=log)
-def _create_and_push_manifest_list(request_id, arches):
+def _create_and_push_manifest_list(request_id, arches, build_tags):
     """
     Create and push the manifest list to the configured registry.
 
     :param int request_id: the ID of the IIB build request
     :param iter arches: an iterable of arches to create the manifest list for
+    :param build_tags: list of extra tag to use for intermediate index image
     :return: the pull specification of the manifest list
     :rtype: str
     :raises IIBError: if creating or pushing the manifest list fails
     """
     buildah_manifest_cmd = ['buildah', 'manifest']
-    output_pull_spec = get_rebuilt_image_pull_spec(request_id)
-    log.info('Creating the manifest list %s locally', output_pull_spec)
-    run_cmd(
-        buildah_manifest_cmd + ['create', output_pull_spec],
-        exc_msg=f'Failed to create the manifest list locally: {output_pull_spec}',
-    )
-    for arch in sorted(arches):
-        arch_pull_spec = _get_external_arch_pull_spec(request_id, arch, include_transport=True)
-        log.debug(
-            'Adding the manifest %s to the manifest list %s', arch_pull_spec, output_pull_spec
-        )
+    output_pull_specs = [get_rebuilt_image_pull_spec(request_id)]
+    if build_tags:
+        conf = get_worker_config()
+        for tag in build_tags:
+            output_pull_specs.append(
+                conf['iib_image_push_template'].format(
+                    registry=conf['iib_registry'], request_id=tag
+                )
+            )
+    for output_pull_spec in output_pull_specs:
```

Comment:
Why does this for loop create a new manifest list for every tag? Can't we simply put this for loop just before line 149? That way we simply push the same manifest list with different tags.

Reply:
yeah, I can improve that